docs/modules/casper.rst

```diff
@@ -688,10 +744,14 @@ Exits phantom with a logged error message and an optional exit status code::
 ``download()``
 -------------------------------------------------------------------------------
 
-**Signature:** ``download(String url, String target[, String method, Object data])``
+**Signature:** ``download(String url, String target[, String method, Object data, Boolean async])``
 
 Saves a remote resource onto the filesystem. You can optionally set the HTTP method using the ``method`` argument, and pass request arguments through the ``data`` object (see `base64encode()`_)::
 
+.. note:: 
+
+since phantom 2.0.0 synchronous downloads are forbidden : By default async is true / For backward compatibilty async is false with phantom 1.x.x::
+
     casper.start('http://www.google.fr/', function() {
```

Comment:
I would never ever change the API behavior (i.e. default values) depending on the engine in use.

Mark this "obsolete / not recommended" in doc, throw a warning to the output when used and advise to use the newly created "waitForDownload" instead.

Better throw an error if used with an engine that does not support it than silently changing behavior here.